src/repo/fs.rs

```diff
@@ -108,12 +109,13 @@ impl BlockStore for FsBlockStore {
 
     async fn remove(&self, cid: &Cid) -> Result<(), Error> {
         let path = block_path(self.path.clone(), cid);
-        let cid = cid.to_owned();
         let cids = self.cids.clone();
         if cids.lock().unwrap().remove(&cid) {
             fs::remove_file(path).await?;
+            Ok(())
+        } else {
+            Err(anyhow!("block not found"))
```

Comment:
Now looking this method as a diff I think that in general with blockstores we have (following the ideas from this great blog post: http://sled.rs/errors.html):

 - IO errors (filesystem broken, underlying media broken, underlying database is broken, background task inaccessible)
 - block removal result
   - not found
   - found but pinned
   - found and removed
   - pending download canceled

So I am thinking something like `Result<Result<(), BlockNotFound>, BadError>`, at least at this stage. At higher level this would probably need to transform to `Result<Result<BlockRemoved, BlockRemovalFailed>, BadError>` when we have the pins in play as well.

Perhaps:

```
enum BlockRemoved {
  Removed(Cid, usize),
  DownloadCancelled(Cid, Duration),
}

struct BlockNotFound(Cid);

enum BlockRemovalFailed {
  Pinned(Cid),
  NotFound(Cid),
}
```

Just thinking out loud for the post-anyhow times.

Reply:
This is great! I will definitely think about this and implement something close to what you have proposed here.

Reply:
Should we return the Cid instead of ()?